### PR TITLE
Ensure interface generation uses package prefixes

### LINKED ISSUE #15154: Java codegen does not respect package prefix for interfaces
See the example project below. It contains one template and one interface and a daml.yaml file that specifies a package prefix.
[foobar.tar.gz](https://github.com/digital-asset/daml/files/9699800/foobar.tar.gz)

If you run `daml codegen java` you end up with the following directory structure:

```
target/generated-sources/
└── iou
    ├── com
    │   └── daml
    │       └── quickstart
    │           ├── iou
    │           │   └── TemplateDecoder.java
    │           └── model
    │               ├── da
    │               │   └── internal
    │               │       └── template
    │               │           └── Archive.java
    │               └── main
    │                   ├── I_Archive.java
    │                   ├── IView.java
    │                   └── T.java
    └── main
        └── I.java
```

so `I.java` just ends up in the wrong place. The package name matches the file location and is also wrong. The reference from `T.java` to `I.java` assumes that the package prefix is right so when you try to compile the code you just end up with an error that `com.daml.quickstart.iou.model.main.I` could not be found.

### PR DESCRIPTION
**Pull request by fayi-da**
Fixes #15154 

CHANGELOG_BEGIN
- [BUGFIX] Interface generation now respects package prefixes argument. This means generated interfaces will now be placed in the right subpackages. 

CHANGELOG_END

### CODE REVIEW

**Comment by ray-roestenburg-da** [comment]
Can you add a test?

File: language-support/java/codegen/src/test/scala/com/digitalasset/daml/lf/codegen/backend/java/ChoiceMetadataFieldsSpec.scala [hunk 1]
```diff
@@ -7,7 +7,7 @@ import com.daml.ledger.javaapi.data.Unit
 import com.daml.ledger.javaapi.data.codegen.ChoiceMetadata
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
-import retro.InterfaceRetro
+import ut.retro.InterfaceRetro
```
Comment (fayi-da) [inline language-support/java/codegen/src/test/scala/com/digitalasset/daml/lf/codegen/backend/java/ChoiceMetadataFieldsSpec.scala:10]: @ray-roestenburg-da My initial thought was that this test now referring to the right package for generated sources was sufficient. I'll see if we have any existing tests for location of files and extend that, else I'll have to write a new suite

File: language-support/java/codegen/src/test/scala/com/digitalasset/daml/lf/codegen/backend/java/InterfaceRetroImplementsSpec.scala [hunk 2]
```diff
@@ -5,7 +5,7 @@ package com.daml.lf.codegen.backend.java
 
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
-import retro.InterfaceRetro
+import ut.retro.InterfaceRetro
```
Comment (fayi-da) [inline language-support/java/codegen/src/test/scala/com/digitalasset/daml/lf/codegen/backend/java/InterfaceRetroImplementsSpec.scala:8]: and this too

**Comment by S11001001** [comment]
> Can you add a specific test that verifies this is now fixed?

@ray-roestenburg-da The test changes on the imports verify that interfaces are in the correct package now; Javapoet decides where to place files in the directory tree directly based on the specified package, and javac enforces that.

**Comment by fayi-da** [comment]
Added a commit to show that we can't run the build if the Interface is in the wrong directory

**Review by S11001001 — APPROVED** [review]

**Pull request merged**